test/base.py

```diff
@@ -18,12 +18,30 @@
 import numpy as np
 
 from qiskit.test import QiskitTestCase
-from qiskit_experiments.framework import ExperimentDecoder, ExperimentEncoder
+from qiskit_experiments.framework import ExperimentDecoder, ExperimentEncoder, ExperimentData
 
 
 class QiskitExperimentsTestCase(QiskitTestCase):
     """Qiskit Experiments specific extra functionality for test cases."""
 
+    def assertSuccess(self, experiment_data: ExperimentData):
+        """Assert that an experiment is succeeded.
+
+        Args:
+            experiment_data: Experiment data to evaluate.
+        """
+        experiment_data.block_for_results()
```

Comment:
There is some subtle async issue where this is returning before the callback statuses seem to get updated properly. I'm not sure how to fix it at the moment, but if you just call block_for_results twice here you should avoid the status == POST PROCESSING failures.
```suggestion
        # Temporary fix to avoid case where block_for_results can return
        # while status still registers as POST PROCESSING is to call it twice
        experiment_data.block_for_results()
        experiment_data.block_for_results()
```